src/gen_rules.ml

```diff
@@ -294,7 +293,11 @@ let gen ~contexts ~build_system
         ~external_lib_deps_mode
         ~stanzas
     in
-    let module M = Gen(struct let sctx = sctx end) in
+    let module P = struct let sctx = sctx end in
+    (* We need to instantiate Install_rules earlier to avoid issues whenever
+     * Super_context is used too soon.
+     * See: https://github.com/ocaml/dune/pull/1354#issuecomment-427922592 *)
```

Comment:
This comment looks good, but I would put it closer to the `module Install_rules = Install_rules.Gen(P)`.

Reply:
Sure